### PR TITLE
sarif-tools: init at 2.0.0

### PR DESCRIPTION
**Pull request by puzzlewolf**
## Description of changes

Add https://github.com/microsoft/sarif-tools, a collection of tools for working with [Static Analysis Results Interchange Format (SARIF)](https://sarifweb.azurewebsites.net/) files.

This is a draft because it requires https://github.com/NixOS/nixpkgs/pull/282347 to fix the failing tests of the `python-docx` dependency (https://github.com/NixOS/nixpkgs/issues/282180). I made sure `sarif-tools` itself is fine by skipping `python-docx`'s tests.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zebreus** [comment]
#282347 is merged, can you rebase this PR against the latest nixpkgs? That should fix the tests.

File: pkgs/development/python-modules/sarif-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    "test_version"
+  ];
+
+  pythonImportsCheck = [
+    "sarif"
+  ];
+
+  meta = with lib; {
+    description = "A set of command line tools and Python library for working with SARIF files";
+    homepage = "https://github.com/microsoft/sarif-tools";
```
Comment (zebreus) [inline pkgs/development/python-modules/sarif-tools/default.nix:63]:
```suggestion
    homepage = "https://github.com/microsoft/sarif-tools";
    changelog = "https://github.com/microsoft/sarif-tools/releases/tag/v${version}";
```

**Review by zebreus — CHANGES_REQUESTED** [review]

**Comment by puzzlewolf** [comment]
@zebreus I rebased and added the changelog link, should be ready for review now :)

**Review by zebreus — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12877,6 +12877,8 @@ with pkgs;
 
   sanjuuni = callPackage ../tools/graphics/sanjuuni { };
 
+  sarif-tools = with python3Packages; toPythonApplication sarif-tools;
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:12880]: write the code below into `pkgs/by-name/sa/sarif-tools/package.nix`:

```nix
{ python3Packages }:

with python3Packages; 
toPythonApplication sarif-tools;
```

and delete the lines in all-packages.

File: pkgs/development/python-modules/sarif-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  disabledTests = [
+    # Broken, re-enable once https://github.com/microsoft/sarif-tools/pull/41 is merged
+    "test_version"
+  ];
+
+  pythonImportsCheck = [
+    "sarif"
+  ];
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/sarif-tools/default.nix:61]:
```suggestion
  meta = {
```

File: pkgs/development/python-modules/sarif-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    description = "A set of command line tools and Python library for working with SARIF files";
+    homepage = "https://github.com/microsoft/sarif-tools";
+    changelog = "https://github.com/microsoft/sarif-tools/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ puzzlewolf ];
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/sarif-tools/default.nix:66]:
```suggestion
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ puzzlewolf ];
```

File: pkgs/development/python-modules/sarif-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    platforms = platforms.all;
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/sarif-tools/default.nix:67]:
```suggestion
```

1. Autoset by Nixpkgs's Python infra
2. Python does not run on all platforms.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/sarif-tools/default.nix (hunk 3 above)
Comment (puzzlewolf) [reply pkgs/development/python-modules/sarif-tools/default.nix:61]: :+1:

File: pkgs/development/python-modules/sarif-tools/default.nix (hunk 4 above)
Comment (puzzlewolf) [reply pkgs/development/python-modules/sarif-tools/default.nix:66]: :+1:

File: pkgs/development/python-modules/sarif-tools/default.nix (hunk 5 above)
Comment (puzzlewolf) [reply pkgs/development/python-modules/sarif-tools/default.nix:67]: Ah, right. I didn't see the note in the [nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#buildpythonpackage-function).

**Comment by puzzlewolf** [comment]
Thanks for the review @AndersonTorres! I applied the changes, should be ready for merging now.

**Review by AndersonTorres — APPROVED** [review]

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 283208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>sarif-tools (python311Packages.sarif-tools)</li>
    <li>sarif-tools.dist (python311Packages.sarif-tools.dist)</li>
    <li>python312Packages.sarif-tools</li>
    <li>python312Packages.sarif-tools.dist</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]
Seems to work!

File: pkgs/development/python-modules/sarif-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "sarif-tools";
+  version = "2.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/sarif-tools/default.nix:23]:
```suggestion
    repo = "sarif-tools";
```

**Pull request merged**